incubator/vault/templates/deployment.yaml

```diff
@@ -171,6 +171,14 @@ spec:
         imagePullPolicy: {{ .Values.vaultExporter.pullPolicy }}
         securityContext:
           readOnlyRootFilesystem: true
+          {{- if .Values.securityContext.usePodSecurityContext }}
+          runAsUser: {{ .Values.securityContext.runAsUser }}
+          {{- if and (.Values.securityContext.runAsUser) (.Values.securityContext.fsGroup) }}
+          {{- if not (eq .Values.securityContext.runAsUser 0.0) }}
+          fsGroup: {{ .Values.securityContext.fsGroup }}
```

Comment:
I'm thinking that it might be safer to have the containers in the pod (and the Vault container itself) run as non-root by default, similar to what the official Hashicorp Vault chart implements:

```yaml
      securityContext:
        runAsNonRoot: true
        runAsGroup: {{ .Values.server.gid | default 1000 }}
        runAsUser: {{ .Values.server.uid | default 100 }}
        fsGroup: {{ .Values.server.gid | default 1000 }}
```

https://github.com/hashicorp/vault-helm/blob/42153168182e6e8eb1aaa4873c39a42d46ea0159/templates/server-statefulset.yaml#L48-L52

The above configuration gives an option to adjust the uid and gip but wouldn't allow the containers to be configured to run as root (which is the default case currently):

https://kubernetes.io/docs/concepts/policy/pod-security-policy/#users-and-groups

```
Pods which have specified neither runAsNonRoot nor runAsUser settings will be mutated to set runAsNonRoot=true, thus requiring a defined non-zero numeric USER directive in the container. No default provided. Setting allowPrivilegeEscalation=false is strongly recommended with this strategy.
```

Are there any reason why we would want Vault to run as root?